commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/servlet/MercanetServlet.java

```diff
@@ -166,26 +181,24 @@ protected void doPost(
 
 				verifyMap.put("Data", data);
 
-				String seal = httpServletRequest.getParameter("Seal");
+				String seal = ParamUtil.getString(httpServletRequest, "Seal");
 
 				verifyMap.put("Seal", seal);
 
-				PaypageResponse paypageResponse = paypageClient.decodeResponse(
-					verifyMap);
+				PaypageResponse paypageResponse =
+					paypageClient.decodeResponse(verifyMap);
 
 				ResponseData responseData = paypageResponse.getData();
 
-				StringBundler transactionReferenceSB = new StringBundler(5);
+				ResponseCode responseCode = responseData.getResponseCode();
```

Comment:
`responseData` and `responseCode` should be sorted _below_ `transactionReferenceSB`. The `response*` variables are only used within the `if` _check_, whereas `transactionReferenceSB` is used in the `if` _block_. If possible, the variables used in the `if` check should come right before the `if` statement.